login_server/player/state/unauthenticated_state.py

```diff
@@ -74,8 +74,9 @@ def handle_login_request(self, request):
             self.player.load()
             self.player.send([
                 a003d()
-                    .set_menu_data(self.class_menu_data)
-                    .set_player(self.player.unique_id, self.player.display_name, '', self.player.loadouts),
+                    .set_menu_data(self.class_menu_data())
+                    .set_player(self.player.unique_id, self.player.display_name, '',
+                                self.player.loadouts, self.player.player_settings.game_setting_mode),
```

Comment:
The mode would become superfluous if we only got the applicable loadouts.

Reply:
This has been simplified now.